src/main/java/seedu/address/model/delivery/Time.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.model.delivery;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a Delivery's time in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidTime(String)}
+ */
+public class Time implements Comparable<Time> {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Time is in minutes, should only contain numbers, and it should be at least 1 digit long, \n"
+                    + "and it should be greater than or equals to 0";
+    public static final String VALIDATION_REGEX = "\\d{1,}";
+    public final String minutes;
+    public final LocalDateTime endTime;
+
+    /**
+     * Constructs a {@code Time}.
+     *
+     * @param minutes A valid time in minutes.
+     * @param endTime A LocalDateTime that the delivery should be done by.
+     */
+    public Time(String minutes, String endTime) {
+        requireNonNull(minutes, endTime);
+
+        LocalDateTime time = LocalDateTime.parse(endTime, DateTimeFormatter.ofPattern("dd MMMM yyyy HH:mm:ss"));
+        this.minutes = minutes;
+        this.endTime = time;
+    }
+
+    /**
+     * @param minutes A valid time in minutes.
+     * @return A new Time, with endTime {@code minutes} from time now.
+     */
+    public static Time timeFromMinutes(String minutes) {
+        checkArgument(isValidTime(minutes), MESSAGE_CONSTRAINTS);
+        long min = Long.parseLong(minutes);
+        LocalDateTime endTime = LocalDateTime.now().plusMinutes(min);
+
+        return new Time(minutes, endTime.format(DateTimeFormatter.ofPattern("dd MMMM yyyy HH:mm:ss")));
```

Comment:
In one of my earlier commits I made that change, for the constructor to take in a LocalDateTime but that implementation had its own problems, such as every single class that needed to use Time had to import LocalDateTime as well as ensure proper formatting of its inputs before passing into the constructor (or else it would throw a DateTimeParseException)